Khan/wonder-blocks
# WB-1200.2: Migrate BirthdayPicker component

## Summary:

- Migrated `BirthdayPicker` from webapp.
- Refactored unit tests to avoid depending on `waitForAnimationFrame`.
- Added Storybook stories and documentation.
- Added `moment` and `jest-date-mock` to the root `package.json`.
- Fixed a couple of Jest warnings on the `wb-dropdown` package. (changed `backgroundColor` to `background`).

Issue: WB-1200

## Test plan:

```
yarn jest packages/wonder-blocks-birthday-picker/
```

Manual testing:

Check the changes in the Storybook instance associated to this PR, or test it locally:

```
yarn start:storybook
```

Navigate to http://localhost:6061/?path=/docs/birthdaypicker--birthday-picker-default

Verify that the stories are rendered correctly and the documentation is correct.
<img width="869" alt="Screen Shot 2022-01-27 at 12 08 24 PM" src="https://user-images.githubusercontent.com/843075/151408544-a4cfdb9e-984f-4298-b275-71233cd9d952.png">

https://user-images.githubusercontent.com/843075/151408556-280425c0-d7e2-49c9-a9ce-4fc81d4c362c.mov
